liquibase-extension-testing/src/main/java/liquibase/extension/testing/util/DownloadUtil.java

```diff
@@ -0,0 +1,80 @@
+package liquibase.extension.testing.util;
+
+import liquibase.Scope;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.logging.Logger;
+import liquibase.util.StreamUtil;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class DownloadUtil {
+
+    private DownloadUtil() {
+
+    }
+
+    public static Path downloadMavenArtifact(String coordinate) {
+        final String[] split = coordinate.split(":");
+        if (split.length != 3) {
+            throw new IllegalArgumentException("Maven coordinates must be in the form groupId:artifactId:version");
+        }
+        return downloadMavenArtifact(split[0], split[1], split[2]);
+
+    }
+
+    public static Path downloadMavenArtifact(String groupId, String artifactId, String version) {
+        final Logger log = Scope.getCurrentScope().getLog(DownloadUtil.class);
+
+        Path path = Paths.get(System.getProperty("user.home"),
+                ".m2",
+                "repository",
+                groupId.replace(".", "/"),
+                artifactId,
+                version,
+                artifactId + "-" + version + ".jar");
+
+        final String url = "https://repo1.maven.org/maven2/" + groupId.replace(".", "/") + "/" + artifactId + "/" + version + "/" + artifactId + "-" + version + ".jar";
+
+        final File pathAsFile = path.toFile();
+        if (pathAsFile.exists()) {
+            log.fine("Artifact " + groupId + ":" + artifactId + ":" + version + " is available at " + path);
+        } else {
+            log.info("Downloading " + url + " to " + path);
+
+            try {
+                final HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
```

Comment:
Might want to use try with resources here

Reply:
HttpURLConnection doesnt implement AutoCloseable so I can't. There is no close method on it even.